File: easy/delete_columns/delete_columns.py
class Solution(object):
    def minDeletionSize(self, A):
        """
        :type A: List[str]
        :rtype: int
        """
        columns_to_delete = 0
        for i in range(0, len(A[0])):
            for j in range(1, len(A)):
                if A[j][i] < A[j-1][i]:
                    columns_to_delete += 1
                    break
        return columns_to_delete


s = Solution()
print("Solution 1 :", s.minDeletionSize(["cba", "daf", "ghi"]))
print("Solution 2 :", s.minDeletionSize(["a", "b"]))
print("Solution 3 :", s.minDeletionSize(["zyx", "wvu", "tsr"]))
print("Solution 4 :", s.minDeletionSize(["rrjk", "furt", "guzm"]))

File: easy/array/delete_columns/delete_columns.go
package main

import "fmt"

func minDeletionSize(A []string) int {
	columnsToDelete := 0
	for i := 0; i < len(A[0]); i++ {
		for j := 1; j < len(A); j++ {
			if A[j][i] < A[j-1][i] {
				columnsToDelete++
				break
			}
		}
	}
	return columnsToDelete
}

func main() {
	fmt.Println("Solution 1 : ", minDeletionSize([]string{"cba", "daf", "ghi"}))
	fmt.Println("Solution 2 : ", minDeletionSize([]string{"a", "b"}))
	fmt.Println("Solution 3 : ", minDeletionSize([]string{"zyx", "wvu", "tsr"}))
}
